### ISSUE TITLE
Fetch version constraints tag does not support paging in tags list

### ISSUE THREAD

**Issue by yorinasub17** [gruntwork-io/fetch#46]
If you are referencing a sufficiently old release tag in a version constraints tag, fetch will fail.

Example:

```
fetch --repo 'https://github.com/gruntwork-io/module-ecs' --tag '~>v0.3.2' foo
```

fails with:

```
ERROR: Error occurred while computing latest tag that satisfies version contraint expression: -1 - Tag does not exist
```

**Comment by josh-padnick** [comment]
Interesting, this definitely sounds like a bug. All the automated tests on this tool assume a small number of releases, so we didn't explicitly test against large numbers of releases. Perhaps when we fix this, for the failing test first, we can find an open source repo that has a large number of releases and triggers this bug.

**Comment by brikis98** [comment]
Could this be a pagination issue? i.e., if there are lots of tags, we are only fetching the first page of results?

**Comment by yorinasub17** [comment]
Yes. Sorry I should have mentioned that in the description, but added that to the title.
